src/mesh/Router.cpp

```diff
@@ -173,6 +173,9 @@ ErrorCode Router::sendLocal(meshtastic_MeshPacket *p, RxSource src)
             handleReceived(p, src);
         }
 
+        p->channel = nodeDB.getNodeChannel(p->to);
```

Comment:
not too familiar with the code, does this override the packet's channel? (ie. client sends packet to node_123 on channel 8 but this changes to NodeInfo.channel)

if so, it is not recommended and packet channel should always be honored. otherwise it can break things (like admin channels).

Reply:
this is a good point. if a channel was requested it should be respected. but, any idea how to differ between not set and primary (0) ?

Reply:
the only packets this is intended for are DM packets, which are up to now on channel 0 - secondary channels will always be non-zero value. So if the channel (from the phone) is zero AND the destination is not a broadcast we can safely alter it. If not, leave it alone.

Reply:
if we want to set this only for DM it might be an idea to not set it in this router but in the client itself? Or somewhere between?

Reply:
NodeInfo.channel set vs. unset is not the issue. overriding the channel will break the special admin/gpio channels and agree clients should pick NodeInfo.channel if that's the intended target. this is best left out.

Reply:
my plan was to check p.channel if a channel was request to not override it :)